src/app/components/annotations/pages/annotationSearchParameters.ts

```diff
@@ -99,38 +156,98 @@ export class AnnotationSearchParameters
   @hasMany<AnnotationSearchParameters, Tag>(TAG, "tags")
   public tagModels?: Tag[];
 
-  public get dateStartedAfter(): DateTime | null {
-    return this.date ? this.date[0] : null;
+  // TODO: use resolvers here once the association resolver decorators return a promise
+  // see: https://github.com/QutEcoacoustics/workbench-client/issues/2148
+  // @hasOne<AnnotationSearchParameters, Project>(PROJECT, "routeProjectId")
+  public routeProjectModel?: Project;
+  // @hasOne<AnnotationSearchParameters, Region>(SHALLOW_REGION, "routeRegionId")
+  public routeRegionModel?: Region;
+  // @hasOne<AnnotationSearchParameters, Site>(SHALLOW_SITE, "routeSiteId")
+  public routeSiteModel?: Site;
+
+  public get recordingDateStartedAfter(): DateTime | null {
+    return this.recordingDate ? this.recordingDate[0] : null;
   }
 
-  public get dateFinishedBefore(): DateTime | null {
-    return this.date ? this.date[1] : null;
+  public get recordingDateFinishedBefore(): DateTime | null {
+    return this.recordingDate ? this.recordingDate[1] : null;
   }
 
-  public get timeStartedAfter(): Duration | null {
-    return this.time ? this.time[0] : null;
+  public get recordingTimeStartedAfter(): Duration | null {
+    return this.recordingTime ? this.recordingTime[0] : null;
   }
 
-  public get timeFinishedBefore(): Duration | null {
-    return this.time ? this.time[1] : null;
+  public get recordingTimeFinishedBefore(): Duration | null {
+    return this.recordingTime ? this.recordingTime[1] : null;
   }
 
+  // TODO: fix up this function
   public toFilter(): Filters<AudioEvent> {
     const tagFilters = filterModelIds<Tag>("tags", this.tags);
-    const filter = this.dateTimeFilters(tagFilters);
+    const dateTimeFilters = this.recordingDateTimeFilters(tagFilters);
+    const routeModelFilter = filterAnd(dateTimeFilters, this.routeFilters());
+    const filter = this.eventDateTimeFilters(routeModelFilter);
     return { filter };
   }
 
   public toQueryParams(): Params {
     return serializeObjectToParams<IAnnotationSearchParameters>(
       this,
-      serializationTable,
+      serializationTable
+    );
+  }
+
+  public modelFilters(): InnerFilter<Project | Region | Site> {
+    if (this.siteModels.length > 0) {
+      return filterModelIds("sites", this.sites);
+    } else if (this.regionModels.length > 0) {
+      return filterModelIds("regions", this.regions);
+    } else {
+      return filterModelIds("projects", this.projects);
+    }
+  }
+
+  public routeFilters(): InnerFilter<AudioEvent> {
+    let siteIds: number[] = [];
+
+    if (this.routeSiteId) {
+      siteIds = [this.routeSiteId];
+    } else if (this.routeRegionId) {
+      siteIds = Array.from(this.routeRegionModel.siteIds);
+    } else {
+      siteIds = Array.from(this.routeProjectModel.siteIds);
+    }
```

Comment:
Audio recordings support filtering by associated site, region, and project models.

Because this data model constructs filter conditions for the audio_events API, our filter condition would have to look something like

```json
{
	"filters": {
		"audio_recordings.projects.id": {
			"in": [1, 2, 3, 4]
		}
	}
}
```

However because this is a filter condition on an association of an association it seems to fail

![image](https://github.com/user-attachments/assets/4c8edc13-e06a-4a14-91db-4c262018d355)

---

Potentially resolved by

https://github.com/QutEcoacoustics/baw-server/issues/689

Reply:
file a separate issue and cross link